src/services/Store/store/helpers.ts

```diff
@@ -145,3 +155,74 @@ export const generateCustomDPath = (dPath: string) => {
   };
   return [path, index];
 };
+
+
+// Handles creating a value transfer for a base asset based on transaction value field
+export const handleBaseAssetTransfer = (
+  valueTransfers: IFullTxHistoryValueTransfer[],
+  value: string,
+  toAddr: TAddress,
+  fromAddr: TAddress,
+  baseAsset: Asset
+): IFullTxHistoryValueTransfer[] => 
+  (valueTransfers.length == 0 && !bigify(value).isZero())
+    ? addBaseAssetValueTransfer(valueTransfers, fromAddr, toAddr, value, baseAsset)
+    : valueTransfers
+
+// Improves verbosity of internal-transaction base value transfers.
+// Only used for incomplete EXCHANGE transaction types (i.e - swapping ERC20 -> ETH)
+export const handleIncExchangeTransaction = (
+  valueTransfers: IFullTxHistoryValueTransfer[],
+  txTypeMetas: ITxMetaTypes,
+  accountsMap: Record<string, boolean>,
+  derivedTxType: string,
+  toAddr: TAddress,
+  fromAddr: TAddress,
+  network: Network
+): IFullTxHistoryValueTransfer[] => 
+  (txTypeMetas[derivedTxType]
+  && txTypeMetas[derivedTxType].type == 'EXCHANGE'
+  && (valueTransfers.filter((t) => accountsMap[generateDeterministicAddressUUID(network.id, t.to)]) || []).length == 0)
+    ? valueTransfers = [...valueTransfers, {
+      asset: generateGenericBase(network.chainId.toString(), network.id),
+      to: toAddr,
+      from: fromAddr,
+      amount: undefined
+    }] : valueTransfers;
+
+export const buildTxHistoryEntry = (networks: Network[], contacts: ExtendedContact[], contracts: ExtendedContract[], assets: ExtendedAsset[], accounts: StoreAccount[]) =>
+  (txTypeMetas: ITxMetaTypes, accountsMap: Record<string, boolean>) =>
+  (tx: ITxReceipt): ITxHistoryEntry => {
+  const network = networks.find(({ id }) => tx.baseAsset.networkId === id) as Network;
```

Comment:
Why the cast?

Reply:
`networks.find(({ id }) => tx.baseAsset.networkId === id)` has implied type of `Network | undefined` but realistically, `find`ing a baseAsset's network will pretty much always result in `Network`, because baseAsset is created from the network itself

Reply:
Maybe use this then
```suggestion
  const network = networks.find(({ id }) => tx.baseAsset.networkId === id)!;
```